qiita_ware/test/test_ebi.py

```diff
@@ -413,63 +226,129 @@ def test_generate_curl_command(self):
                                                test_ebi_access_key,
                                                test_ebi_skip_curl_cert,
                                                test_ebi_dropbox_url)
-        exp_with_cert = ('curl '
-                         '-F "SUBMISSION=@submission.xml" '
-                         '-F "STUDY=@study.xml" '
-                         '-F "SAMPLE=@sample.xml" '
-                         '-F "RUN=@run.xml" '
-                         '-F "EXPERIMENT=@experiment.xml" '
-                         '"ebi_dropbox_url/?auth=ERA%20ebi_seq_xfer_user'
-                         '%20ebi_access_key%3D"')
-        self.assertEqual(obs, exp_with_cert)
+        exp = ('curl '
+               '-F "SUBMISSION=@%(xml_dir)ssubmission.xml" '
+               '-F "STUDY=@%(xml_dir)sstudy.xml" '
+               '-F "SAMPLE=@%(xml_dir)ssample.xml" '
+               '-F "RUN=@%(xml_dir)srun.xml" '
+               '-F "EXPERIMENT=@%(xml_dir)sexperiment.xml" '
+               '"ebi_dropbox_url/?auth=ERA%%20ebi_seq_xfer_user'
+               '%%20ebi_access_key%%3D"') % {'xml_dir': submission.xml_dir}
+        self.assertEqual(obs, exp)
 
 
 @qiita_test_checker()
 class TestEBISubmissionWriteRead(TestEBISubmission):
-    def test_generate_demultiplexed_fastq(self):
-        # generating demux file for testing
-        if not isdir(self.temp_dir):
-            makedirs(self.temp_dir)
-
+    def write_demux_files(self, prep_template):
+        """Writes a demux test file to avoid duplication of code"""
         fna_fp = join(self.temp_dir, 'seqs.fna')
-        self.files_to_remove.append(fna_fp)
-        # demux_fp doesn't need to be removed cause is being moved to be
-        # the ppd being created and will be removed there
         demux_fp = join(self.temp_dir, 'demux.seqs')
         with open(fna_fp, 'w') as f:
             f.write(FASTA_EXAMPLE)
         with File(demux_fp, "w") as f:
             to_hdf5(fna_fp, f)
 
-        # Adding a new PreprocessedData to study 1
-        # The 3rd parameter 1 is the PreprocessedIlluminaParams id, which is
-        # not relevant for this test
+        ppd = PreprocessedData.create(Study(1),
+                                      "preprocessed_sequence_illumina_params",
+                                      1, [(demux_fp, 6)], prep_template)
+
+        return ppd
+
+    def generate_new_prep_template_and_write_demux_files(self):
+        """Creates a new prep template and writes demux test file to avoid
+        duplication of code"""
+
+        # ignoring warnings generated when adding templates
+        simplefilter("ignore")
```

Comment:
I'm just trying to be safer - if we know that a warning should be raised here, we should check that is raised. If any future change in the code makes that warning to not be raised I'm not sure if that can break some of the functionality here...